### PR TITLE
Widen integer literal to unsigned when required

### LINKED ISSUE #515: Integer literals incorrectly treated as `Clong`
I observed that macro definitions translated like the following:

```c
#define TEST 0x80000001L
```
to
```julia
const TEST = Clong(0x80000001)
```

fail with an `ERROR: InexactError: convert(Int32, 0x80000001)` because we are wrapping the literal with `Clong` (`:= Int32`). AFAIK C compilers will treat the literal as a `Culong` (`:= UInt32`), detecting that the literal is too large for a `Clong`. The solution would be to apply the same rule, and in this case, wrap with a `Culong`.

Any pointers as to where to perform the fix?

### PR DESCRIPTION
**Pull request by serenity4**
Fixes #515.

This include an additional fix with `txt_no_sfx = replace(txt_no_sfx, r"^0X" => "0x")` to also support integer constants formatted as `0X012345678`, which otherwise throw an error when executed.

### CODE REVIEW

File: src/generator/macro.jl [hunk 1]
```diff
@@ -66,7 +66,11 @@ function c_non_float_literal_to_julia(sfx)
     elseif occursin("ul", sfx) || occursin("lu", sfx)
         return "Culong"
     elseif !unsigned && endswith(sfx, "l")
-        return "Clong"
+        if length(txt_no_sfx) == 10 && startswith(txt_no_sfx, r"0x[8-9a-fA-F]")
```
Comment (JamesWrigley) [inline src/generator/macro.jl:69]: I think it might be simpler to try parsing the string as a `Clong` and falling back to `Culong` if that fails.

**Review by JamesWrigley — COMMENTED** [review]
Would be nice to make a changelog entry too :)

File: src/generator/macro.jl (hunk 1 above)
Comment (serenity4) [reply src/generator/macro.jl:69]: True, it would also help with non-hexadecimal representations that may still represent large numbers.

**Comment by serenity4** [comment]
Nightly failures seem unrelated, locally the fix handles hexadecimal and decimal literal integers correctly (using `Clong` by default and only wrap in `Culong` when the value doesn't fit).

**Comment by JamesWrigley** [comment]
Sorry forgot to mention, could we add a test too? I would recommend adding a header file in https://github.com/JuliaInterop/Clang.jl/tree/master/test/include and then using it in https://github.com/JuliaInterop/Clang.jl/blob/master/test/generators.jl.

**Comment by serenity4** [comment]
Sure, thanks for the pointer! Should be good now.

**Comment by serenity4** [comment]
All good :)

**Comment by JamesWrigley** [comment]
Unofficial LGTM!

**Comment by Gnimuc** [comment]
Thanks！

**Pull request merged**